### PR TITLE
Dr/purge fortran

### PR DESCRIPTION
**Pull request by ravil-mobile**
Purged the old Fortran code from dr/cpp branch

### CODE REVIEW

File: src/Solver/Interoperability.cpp [hunk 1]
```diff
@@ -224,15 +219,14 @@ extern "C" {
...
 		  int* outputGroups, int outputGroupsSize,
 		  double freeSurfaceInterval, const char* freeSurfaceFilename, const char* xdmfWriterBackend,
       const char* receiverFileName, double receiverSamplingInterval, double receiverSyncInterval,
       bool isPlasticityEnabled, bool isEnergyTerminalOutputEnabled, double energySyncInterval) {
       auto outputGroupBounds = std::unordered_set<int>(outputGroups, outputGroups + outputGroupsSize);
-    e_interoperability.initializeIO(mu, slipRate1, slipRate2, slip, slip1, slip2, state, strength,
+    e_interoperability.initializeIO(
                                     numSides, numBndGP, refinement, outputMask, plasticityMask, outputRegionBounds,
                                     outputGroupBounds,
                                     freeSurfaceInterval, freeSurfaceFilename, xdmfWriterBackend,
```
Comment (ravil-mobile) [inline src/Solver/Interoperability.cpp:232]: a strange indentation

**Comment by sebwolf-de** [comment]
I did most of the work, so it feels strange to review this PR :see_no_evil:

File: src/Solver/Interoperability.cpp [hunk 2]
```diff
@@ -896,10 +886,33 @@ seissol::Interoperability::initializeIO(double* mu, double* slipRate1, double* s
   
 	// Initialize checkpointing
 	int faultTimeStep;
-	bool hasCheckpoint = seissol::SeisSol::main.checkPointManager().init(reinterpret_cast<real*>(m_ltsTree->var(m_lts->dofs)),
+
+  // Only R&S friction explicitly stores the state variable, otherwise use the accumulated slip magnitude
```
Comment (Thomas-Ulrich) [inline src/Solver/Interoperability.cpp:890]: why not 0?
In the end, accumulatedSlipMagnitude will be stored anyway.

File: src/Reader/readpar.f90 [hunk 3]
```diff
@@ -978,129 +797,20 @@ SUBROUTINE readdr(IO, EQN, DISC, BND, IC, MPI)
      logError(*) 'Dynamic rupture model file "', trim(ModelFileName), '" does not exist.'
      call exit(134)
     endif
-    !
-    DISC%DynRup%ModelFileName = ModelFileName
```
Comment (Thomas-Ulrich) [inline src/Reader/readpar.f90:982]: should that be removed?

File: src/Reader/readpar.f90 [hunk 4]
```diff
@@ -978,129 +797,20 @@ SUBROUTINE readdr(IO, EQN, DISC, BND, IC, MPI)
...
-             EQN%RS_sv0 = RS_sv0
-           CASE DEFAULT
-             logError(*) 'Unknown Stress Background Type: ',DISC%DynRup%BackgroundType
-             call exit(134)
-           END SELECT
+    !FRICTION LAW CHOICE
+    EQN%FL = FL
+    EQN%GPwise = GPwise
+    EQN%refPointMethod = refPointMethod
+    EQN%XRef = XRef
```
Comment (Thomas-Ulrich) [inline src/Reader/readpar.f90:805]: Is it not read with:
```double xref = getWithDefault(yamlDrParams, "xref", 0.0);```?

File: src/Reader/readpar.f90 [hunk 5]
```diff
@@ -978,129 +797,20 @@ SUBROUTINE readdr(IO, EQN, DISC, BND, IC, MPI)
...
-           ELSEIF(DISC%DynRup%OutputPointType.EQ.5) THEN
-                ! ALICE: TO BE DONE
-                ! fault receiver + 2 dimensional fault output
-                call readpar_faultElementwise(EQN,BND,IC,DISC,IO,CalledFromStructCode)
-                call readpar_faultAtPickpoint(EQN,BND,IC,DISC,IO,MPI,CalledFromStructCode)
-           ELSE
-               logError(*) 'Unkown fault output type (e.g.3,4,5)',DISC%DynRup%OutputPointType
-               call exit(134)
-           ENDIF ! DISC%DynRup%OutputPointType
+    IF (EQN%GPwise .EQ.1) THEN
```
Comment (Thomas-Ulrich) [inline src/Reader/readpar.f90:809]: is it still a relevant parameter?

**Comment by sebwolf-de** [comment]
Unfortunately, it does not fix 2 in #48 :/

File: src/Reader/readpar.f90 (hunk 4 above)
Comment (sebwolf-de) [reply src/Reader/readpar.f90:805]: Good point!

File: src/Reader/readpar.f90 (hunk 3 above)
Comment (sebwolf-de) [reply src/Reader/readpar.f90:982]: Yes, we initialize everything in Fortran, so there is no need to store the model file name in Fortran.

File: src/Solver/Interoperability.cpp (hunk 2 above)
Comment (sebwolf-de) [reply src/Solver/Interoperability.cpp:890]: Then, I would need to create an array with a lot of zeros first ;-)

File: src/Reader/readpar.f90 (hunk 5 above)
Comment (sebwolf-de) [reply src/Reader/readpar.f90]: No, its not. Also removed it.

File: src/Solver/Interoperability.cpp (hunk 2 above)
Comment (Thomas-Ulrich) [reply src/Solver/Interoperability.cpp:890]: Ok, in the end, we would need some more flexibility for checkpointing, e.g. with TP T and P should be added, etc.
Let's leave it like that now, or add a TODO

**Review by Thomas-Ulrich — APPROVED** [review]
LGTM

**Pull request merged**